consopt.go

```diff
@@ -97,10 +101,11 @@ func WithSize(size ...int) ConsOpt {
 			l.dims = size[0]
 			return l, nil
 		case sizeSetter:
-			l.SetSize(size[0])
-			return layer, nil
+			return layer, l.SetSize(size[0])
```

Comment:
does this work? `SetSize` presumably mutates the data in `layer`. But when it's returned at the same level...

Reply:
if SetSize returns an error then I guess the layer is ignored and the error is handled, I can handle the error independently to return an nil, err